javascript/ql/src/Security/CWE-020/IncompleteUrlSubstringSanitization.ql

```diff
@@ -34,33 +44,22 @@ where
     // target is a HTTP URL to a domain on any TLD
     target.regexpMatch("(?i)https?://([a-z0-9-]+\\.)+([a-z]+)(:[0-9]+)?/?")
   ) and
+  (
+    if check instanceof StringOps::StartsWith
+    then msg = "can be the beginning of the URL, and arbitrary hosts may come after it"
```

Comment:
Perhaps simplify to "may be followed by an arbitrary host name". I would expect it to be clear from context that the string can appear at the beginning of the URL, and that isn't the problem we're flagging.